### DISCUSSION TITLE
Where are structurizr-extensions gone ?

### DISCUSSION THREAD

**Discussion by Riduidel** [structurizr/java#294] [Q&A]
@simonbrowndotje There used to be a structurizr-extensions repository containing the structurizr-plantuml exporter and some tools to detect elements in code. The repository is now archived. Can we discuss that subject ? (cause there are some evolutions I would like to do to structurizr-plantuml, whcih seems now impossible)

**Comment by simonbrowndotje** [comment]
> (cause there are some evolutions I would like to do to structurizr-plantuml, whcih seems now impossible)

The repo is archived, but you're welcome to fork it.

**Comment by carljonescf** [comment]
Understand the reasons for deprecation. Wonder what the current view is on the use of the component model?

I still see recommendations to use reflection on the site, but also notice that a number of demos stop at the container model (or groups of containers).

Is best practice to look at using reflection, to manual update the component model or to stop at the container model? 

(I'm guessing that a "it depends" response may be coming :-) )

Thanks

**Comment by simonbrowndotje** [comment]
> Is best practice to look at using reflection, to manual update the component model or to stop at the container model?

The vast majority of teams (perhaps upwards of 95%) only use the system context and container diagrams. And my recommendation for teams who want to retrospectively model components is to do so via reverse-engineering (e.g. reflection, static analysis, etc).

The reason for the archival of the `structurizr-analysis` library is that it needs updating to support the latest version of the Structurizr libraries and Java versions. I have started rewriting it, but it's very low down my todo list since there is almost no demand for it and I have limited time to work for free.

**Reply by carljonescf** [reply]
Thanks Simon. Understood.

I'm currently looking at how to combine C4 with Spring Modulith in teaching (very early days).

**Comment by Riduidel** [comment]
In fact I also have a low-priority project to take hold on that part of
structurizr, but the lack of time is not helping

Le 23/07/2024 à 10:17, Carl Jones a écrit :

**Comment by simonbrowndotje** [comment]
I have added an initial rewrite of the `structurizr-analysis` library to the repo, which can be found at https://github.com/structurizr/java/tree/master/structurizr-component

It's functional, but there are no tests yet, and you will need to build from source to use it. An example can be found at https://github.com/structurizr/java/blob/master/structurizr-component/src/test/java/com/structurizr/component/example/Example.java
